Source/Diagnostics/FieldIO.cpp

```diff
@@ -352,6 +352,31 @@ AverageAndPackVectorField( MultiFab& mf_avg,
     }
 }
 
+/** \brief Takes the specified component of the three fields and
+ * averages and packs them into the MultiFab mf_avg.
+ */
+void
+AverageAndPackVectorFieldComponents (MultiFab& mf_avg,
+                                     const std::array< std::unique_ptr<MultiFab>, 3 >& vector_field,
+                                     const DistributionMapping& dm,
+                                     int& dcomp, const int ngrow )
+{
+    if (vector_field[0]->nComp() > 1) {
+        std::array<std::unique_ptr<MultiFab>,3> vector_field_component;
+        vector_field_component[0].reset(new MultiFab(vector_field[0]->boxArray(), dm, 1, vector_field[0]->nGrowVect()));
+        vector_field_component[1].reset(new MultiFab(vector_field[1]->boxArray(), dm, 1, vector_field[1]->nGrowVect()));
+        vector_field_component[2].reset(new MultiFab(vector_field[2]->boxArray(), dm, 1, vector_field[2]->nGrowVect()));
+        for (int icomp=0 ; icomp < vector_field[0]->nComp() ; icomp++) {
+            // This is a bit inefficient since it copies the data. Can this be done with pointers?
```

Comment:
maybe @WeiqunZhang knows this.

Reply:
You can do

```
vector_field_component[0].reset(new MultiFab(*vector_field[0], amrex::make_alias, icomp, 1));
..[1]...
...[2]...
AverageAndPackVectorField(...)
```
without deep copy.

Reply:
@RemiLehe @dpgrote did you see this comment?

Reply:
Yep. I'll make the change.